module/zcommon/zfeature_common.c

```diff
@@ -333,6 +334,13 @@ zpool_feature_init(void)
 	    ZFEATURE_FLAG_READONLY_COMPAT | ZFEATURE_FLAG_PER_DATASET,
 	    project_quota_deps);
 	}
+
+	{
+	zfeature_register(SPA_FEATURE_ALLOCATION_CLASSES,
+	    "com.intel:allocation_classes", "allocation_classes",
+	    "Support for separate allocation classes.",
```

Comment:
not sure this is accurate anymore. the original proposed changes would sastisfy this description, but as of now it's just a single special allocation class.

Reply:
@kpande - a dedup class is planned